### ISSUE TITLE
Suggestion: Respect Mod Dependencies when sorting the mods

### ISSUE THREAD

**Issue by lisanhu** [RimSort/RimSort#260]
**Is your feature request related to a problem? Please describe.**
Currently, the sorting function only respects the <loadAfter> labels in the About.xml files in the mods. Sometimes, when my mod's name is alphabetically before a dependent mod, if I don't explicitly set up loadAfter in the About.xml, it will be sorted before its dependent.

**Describe the solution you'd like**
Is it possible to also respect the mod dependencies in the sorting function?

**Describe alternatives you've considered**

**Additional context**

**Comment by Limfjorden** [comment]
How would rimsort be able to know the mod should be loaded after, if you don't specify it?

If you mean the steam dependencies, you can help making it work by updating the Steam DB. It indexes steam dependencies.

**Comment by lisanhu** [comment]
There's a <modDependencies> label in the About.xml files for the mods. This label specifies what mods are required to load before this mod, and it is currently not respected. Steam dependencies could be different from this and this is usually the strongest restriction.

**Comment by LionelColaso** [comment]
can you please test the update branch of rimsort and see if its still a issue ?

**Comment by lisanhu** [comment]
I have pulled the Feb 12 update branch, and compiled and ran it on my system (Pop OS 22.04). The problem still exists. Actually, I just realized this is related to #259 . 

In [NL] Facial Animation -- Experimentals, the About.xml has the following lines:
```
<modDependencies>
      <li>
        <packageId>Nals.FacialAnimation</packageId>
        <displayName>[NL] Facial Animation - WIP</displayName>
        <steamWorkshopUrl>https://steamcommunity.com/sharedfiles/filedetails/?id=1635901197</steamWorkshopUrl>
      </li>
    </modDependencies>
    <loadAfter>
      <li>Nals.FacialAnimation</li>
    </loadAfter>
```

However, after sorting, I still got this:
![image](https://github.com/RimSort/RimSort/assets/3086096/c0830bd7-6346-41cd-8a0e-0c911554f627)

Facial Animation - WIP is still after Experimentals

**Comment by LionelColaso** [comment]
@lisanhu can you check if you have the same issue with update branch the issue must be fixed 

https://github.com/RimSort/RimSort/tree/update

**Comment by lisanhu** [comment]
Thanks for your quick update and timely reply! The problem still exists for me. I tried to minimize the mod list and export it to clipboard.

```
Created with RimSort Alpha-v1.0.6.2-hf
RimWorld game version this list was created for: 1.4.3901 rev238
Total # of mods: 6

RimWorld [ludeon.rimworld][https://store.steampowered.com/app/294100/RimWorld]
[NL] Facial Animation - Experimentals [nals.facialanimationexperimentals][https://steamcommunity.com/sharedfiles/filedetails/?id=2581693737]
Humanoid Alien Races [erdelf.humanoidalienraces][https://github.com/erdelf/AlienRaces/wiki]
Epona race Renaissance [epona.eponarenaissance][https://steamcommunity.com/sharedfiles/filedetails/?id=2624033197]
[NL] Facial Animation - WIP [nals.facialanimation][https://steamcommunity.com/sharedfiles/filedetails/?id=1635901197]
Epona Facial Animation [ancot.eponafacialanimation][https://steamcommunity.com/sharedfiles/filedetails/?id=2873089116]
```
![image](https://github.com/RimSort/RimSort/assets/3086096/d948184b-4a02-405e-8f2b-cf9e45684fe1)

Really appreciate your help!

**Comment by LionelColaso** [comment]
@lisanhu sorting works correctly if you do not use community_database 
when using community_database  i can replicate the issue u have 
i think there may be sort order issue with community_database 
i will check community_database and fix it later you too can do it by searching the id
below screenshot one with community_database and one without
![image](https://github.com/RimSort/RimSort/assets/2488703/64678c03-8154-4c82-b895-2c69c9577191)
![image](https://github.com/RimSort/RimSort/assets/2488703/19cff128-3209-446e-a2ba-02d84660d997)

**Comment by lisanhu** [comment]
I can confirm the issue is fixed for me using no database file. Thanks a lot!
